gui/wxpython/core/settings.py

```diff
@@ -57,9 +67,12 @@ def color(item):
 
 def settings_JSON_decode_hook(obj):
     """Decode hex color saved in settings into tuple"""
+    def colorhex2tuple(hexcode):
+        return tuple(int(hexcode.lstrip('#')[i:i + 2], 16) for i in range(0, len(hexcode) - 2, 2))
```

Comment:
IIUC `lstrip('#')` is called 3 or 4 times. Perhaps an unnecessary one-liner. A version with variable for `hexcode.lstrip('#')` is more readable and results in a significantly shorter line (the `+ 2` is no longer needed). There is no performance hit for that (actually, it is slightly faster).

```python
>>> def colorhex2tuple(hexcode):
...     return tuple(int(hexcode.lstrip('#')[i:i + 2], 16) for i in range(0, len(hexcode) - 2, 2))
...
>>> def colorhex2tuple2(hexcode):
...     numbers = hexcode.lstrip('#')
...     return tuple(int(numbers[i:i + 2], 16) for i in range(0, len(numbers), 2))
... 
>>> timeit.timeit(lambda: colorhex2tuple("#20FF41"), number=1000000)
1.4110316660953686
>>> timeit.timeit(lambda: colorhex2tuple2("#20FF41"), number=1000000)
1.280416122986935
>>> colorhex2tuple2("#20FF41")
(32, 255, 65)
>>> colorhex2tuple2("#20FF41FE")
(32, 255, 65, 254)
```